app/views/admin/collections/show.html.erb

```diff
@@ -57,12 +57,14 @@ Unless required by applicable law or agreed to in writing, software distributed
         <% end %>
       </div>
       <div class="collection_poster">
+        <% is_depositor=can?(:read, Admin::Collection) && !can?(:manage, Admin::Collection) %>
         <%= form_for @collection, url: attach_poster_admin_collection_path(@collection.id), html: {method: "post"} do |form| %>
           <%= form.file_field :poster, id: 'poster_input', class: "filedata", style: "height:0px;width:0px;" %>
-          <input type="button" class="btn btn-primary btn-xs" onclick="$('.filedata').click();" value="Upload Poster" />
+          <input type="button" class="btn btn-primary btn-xs" onclick="$('.filedata').click();" value="<% if @collection.poster.present? %>Change Poster<%else%>Upload Poster<%end%>"
+            <% if is_depositor %>disabled<%end%> />
```

Comment:
I think these authorization checks would be better if they were scoped to the specific collection and action being taken:
```suggestion
            <% if cannot?(:edit, @collection) %>disabled<%end%> />
```